src/screens/FeedScreen/ContentLink.js

```diff
@@ -57,53 +58,74 @@ const ContentLink = ({
+      tab: DISCOVERY_TAB_USERS
+    });
+  };
+
   return (
-    <View style={styles.contentFeed(hasComment)}>
-      <TouchableNativeFeedback
-        disabled={isTouchableDisabled}
-        onPress={onPress}
-        testID="contentLinkContentPressable">
-        <>
-          <TouchableWithoutFeedback onPress={onPress}>
-            {renderMessageContentLink()}
-          </TouchableWithoutFeedback>
-          {smartRender(Card, {
-            domain: og.domain,
-            date: new Date(og.date).toLocaleDateString(),
-            domainImage: og.domainImage,
-            title: og.title,
-            description: og.description,
-            image: og.image,
-            url: og.url,
-            onHeaderPress,
-            onCardContentPress,
-            score,
-            item,
-            heightTopic,
-            textHeight,
-            contentHeight
-          })}
-        </>
-      </TouchableNativeFeedback>
-      <TopicsChip
-        onLayout={handleTopicLayout}
-        topics={topics}
-        fontSize={normalizeFontSize(14)}
-        text={sanitizeUrl}
-        // topicContainer={styles.topicStyle}
-      />
-    </View>
+    <BlurredLayer
+      withToast={true}
+      onPressContent={handleBlurredContent}
+      isVisible={isBlurredPost}
+      containerStyle={{
+        flex: 1
+      }}>
+      <View style={styles.contentFeed}>
+        <TouchableNativeFeedback
+          disabled={isTouchableDisabled}
+          onPress={onPress}
+          testID="contentLinkContentPressable"
+          style={{flex: 1}}>
+          <>
+            <TouchableWithoutFeedback onPress={onPress}>
+              {renderMessageContentLink()}
+            </TouchableWithoutFeedback>
+            {smartRender(Card, {
+              domain: og.domain,
+              date: new Date(og.date).toLocaleDateString(),
+              domainImage: og.domainImage,
+              title: og.title,
+              description: og.description,
+              image: og.image,
+              url: og.url,
+              onHeaderPress,
+              onCardContentPress,
+              score,
+              item,
+              heightTopic,
+              textHeight,
+              contentHeight,
+              containerStyle: {flex: topics.length > 0 ? 0 : 1}
+            })}
+          </>
+        </TouchableNativeFeedback>
+        {!isBlurredPost && (
+          <TopicsChip
+            onLayout={handleTopicLayout}
+            topics={topics}
+            fontSize={normalizeFontSize(14)}
+            text={sanitizeUrl}
+            topicContainer={styles.topicStyle}
+            topicItemStyle={styles.topicItemStyle}
+          />
+        )}
+      </View>
+    </BlurredLayer>
   );
 };
 
 export default ContentLink;
 
 const styles = StyleSheet.create({
-  contentFeed: (hasComment) => ({
-    flex: hasComment ? 1 : 0,
+  contentFeed: {
+    flex: 1,
     marginHorizontal: 6,
     backgroundColor: COLORS.white
-  }),
+  },
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `BlurredLayer` component is introduced to handle the blurred posts. However, it seems like the navigation destination when the blurred content is pressed is hardcoded to 'DiscoveryScreen' with tab as 'DISCOVERY_TAB_USERS'. This might not always be the desired behavior and could lead to unexpected user experience. Consider making this a prop or some configurable option.

```diff
  const handleBlurredContent = () => {
    navigation.navigate('DiscoveryScreen', {
      tab: DISCOVERY_TAB_USERS
    });
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->